compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt

```diff
@@ -0,0 +1,274 @@
+         */
+        override fun insertText(text: String) {
+            getCursorPos()?.let {
+                _tempCursorPos = it + text.length
+            }
+            sendEditCommand(CommitTextCommand(text, 1))
+        }
+
+        /**
+         * Deletes a character from the displayed text.
+         * Remove the character just before the cursor from your class’s backing store and redisplay the text.
+         * https://developer.apple.com/documentation/uikit/uikeyinput/1614572-deletebackward
+         */
+        override fun deleteBackward() {
+            sendEditCommand(
+                BackspaceCommand()
+            )
+        }
+
+        /**
+         * The text position for the end of a document.
+         * https://developer.apple.com/documentation/uikit/uitextinput/1614555-endofdocument
+         */
+        override fun endOfDocument(): Long = getState()?.text?.length?.toLong() ?: 0L
+
+        /**
+         * The range of selected text in a document.
+         * If the text range has a length, it indicates the currently selected text.
+         * If it has zero length, it indicates the caret (insertion point).
+         * If the text-range object is nil, it indicates that there is no current selection.
+         * https://developer.apple.com/documentation/uikit/uitextinput/1614541-selectedtextrange
+         */
+        override fun getSelectedTextRange(): SkikoTextRange? {
+            val cursorPos = getCursorPos()
+            if (cursorPos != null) {
+                return SkikoTextRange(cursorPos, cursorPos)
+            }
+            val selection = getState()?.selection
+            return if (selection != null) {
+                SkikoTextRange(selection.start, selection.end)
+            } else {
+                null
+            }
+        }
+
+        override fun setSelectedTextRange(range: SkikoTextRange?) {
+            if (range != null) {
+                sendEditCommand(
+                    SetSelectionCommand(range.start, range.end)
+                )
+            }
+        }
+
+        override fun selectAll() {
+            sendEditCommand(
+                SetSelectionCommand(0, endOfDocument().toInt())
+            )
+        }
+
+        /**
+         * Returns the text in the specified range.
+         * https://developer.apple.com/documentation/uikit/uitextinput/1614527-text
+         * @param range A range of text in a document.
+         * @return A substring of a document that falls within the specified range.
+         */
+        override fun textInRange(range: SkikoTextRange): String? {
+            val text = getState()?.text
+            return if (!text.isNullOrEmpty() && range.start >= 0 && range.end >= 0 && text.length >= range.end) {
+                val substring = text.substring(range.start, range.end)
+                substring.replace("\n", "").ifEmpty { null } //todo maybe redundant
```

Comment:
Is there a reason to replace and return null if empty? If not, better to return text as it is.

Reply:
Yeah, we can return empty String ""